splice_encoding/src/main/java/com/splicemachine/utils/logging/MaskPatternLayout.java

```diff
@@ -1,23 +1,71 @@
 package com.splicemachine.utils.logging;
 
+import com.splicemachine.utils.Pair;
 import org.apache.log4j.Logger;
 import org.apache.log4j.PatternLayout;
 import org.apache.log4j.spi.LoggingEvent;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 public class MaskPatternLayout extends PatternLayout {
 
-    private static final String MASK = "***MASKED SENSITIVE INFO***";
+    private String maskString = "***MASKED SENSITIVE INFO***";
     private Pattern maskPattern = null;
+    private Logger logger = Logger.getLogger(this.getClass());
 
     public String getMaskPattern() {
         return maskPattern.toString();
     }
 
     public void setMaskPattern(String pattern) {
-        maskPattern = Pattern.compile(pattern);
+        try {
+            maskPattern = Pattern.compile(pattern);
+        } catch (Exception e) {
+            logger.error("Error to compile regex for mask pattern, will not mask anything", e);
+        }
+    }
+
+    public void setMaskString(String maskString) {
+        this.maskString = maskString;
+    }
+
+    public String getMaskString() {
+        return this.maskString;
+    }
+
+    public static String maskMessage(String message, Pattern maskPattern, String maskString) {
```

Comment:
Why is this method `static`?

Reply:
This method has no need to be non-static and this also makes it easy to test.

Reply:
Two out of three arguments are instance variables so I think that's what makes the method an instance method.